### PR TITLE
HPCC-18904 Various DESDL U/I changes

### PR DESCRIPTION
**Pull request by miguelvazq**
- Alphabetize definition items
- By default land on Dynamic ESDL tab

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by miguelvazq** [comment]
@GordonSmith Please review.

File: esp/src/eclwatch/DynamicESDLDefinitionQueryWidget.js [hunk 1]
```diff
@@ -94,11 +95,14 @@ define([
...
                     });
+                    Utility.alphanumSort(results, "Name");
                 }
                 context.store.setData(results);
                 context.grid.refresh();
-                var firstRowSelection = context.store.query({Id: 0});
-                context.grid.select({Name:firstRowSelection[0].Name});
+                if (context.params.firstLoad) {
+                    var firstRowSelection = context.store.query({Id: results[0].Id});
+                    context.grid.select({Name:firstRowSelection[0].Name});
```
Comment (GordonSmith) [inline esp/src/eclwatch/DynamicESDLDefinitionQueryWidget.js]: Need to test if firstRowSelection.length?

File: esp/src/eclwatch/DynamicESDLQueryWidget.js [hunk 2]
```diff
@@ -57,6 +57,15 @@ define([
             this.detailsWidget.widget._Binding.set("disabled", true);
         },
 
+        initTab: function () {
+            var currSel = this.getSelectedChild();
+            if (currSel.id == this.definitionQueryWidget.id && !this.definitionQueryWidget.initalized) {
```
Comment (GordonSmith) [inline esp/src/eclwatch/DynamicESDLQueryWidget.js]: Should be "==="

**Review by GordonSmith — CHANGES_REQUESTED** [review]

**Comment by richardkchapman** [comment]
@miguelvazq

**Comment by miguelvazq** [comment]
@GordonSmith Modified.

File: esp/src/eclwatch/DynamicESDLDefinitionQueryWidget.js [hunk 3]
```diff
@@ -94,11 +95,16 @@ define([
                         }
                         results.push(Def);
                     });
+                    Utility.alphanumSort(results, "Name");
                 }
                 context.store.setData(results);
                 context.grid.refresh();
-                var firstRowSelection = context.store.query({Id: 0});
-                context.grid.select({Name:firstRowSelection[0].Name});
+                if (context.params.firstLoad) {
```
Comment (GordonSmith) [inline esp/src/eclwatch/DynamicESDLDefinitionQueryWidget.js]: Should probably be `context.params.firstLoad && ressults.length`

**Review by GordonSmith — CHANGES_REQUESTED** [review]

**Comment by miguelvazq** [comment]
Good point. Updated.

**Comment by richardkchapman** [comment]
@GordonSmith Ok to merge?

**Comment by GordonSmith** [comment]
Looks good.

**Pull request merged**